session/session.go

```diff
@@ -1627,14 +1628,16 @@ func (s *session) isTxnRetryable() bool {
 func (s *session) NewTxn(ctx context.Context) error {
 	if s.txn.Valid() {
 		txnID := s.txn.StartTS()
+		txnScope := s.txn.Scope()
 		err := s.CommitTxn(ctx)
 		if err != nil {
 			return err
 		}
 		vars := s.GetSessionVars()
 		logutil.Logger(ctx).Info("NewTxn() inside a transaction auto commit",
 			zap.Int64("schemaVersion", vars.TxnCtx.SchemaVersion),
-			zap.Uint64("txnStartTS", txnID))
+			zap.Uint64("txnStartTS", txnID),
+			zap.String("txnScope", txnScope))
 	}
 
 	txn, err := s.store.Begin()
```

Comment:
I went through the usage of `NewTxn` and I think it should also apply `CheckAndGetTxnScope`.